cypress/integration/models/stakeholders.ts

```diff
@@ -22,53 +37,111 @@ export class Stakeholders {
+            removeMember(group);
+        });
+    }
+
     getStakeholderName(): string {
-        this.stakeholderName = faker.name.findName();
-        return this.stakeholderName;
+        return faker.name.findName();
     }
 
     getStakeholderEmail(): string {
-        this.stakeholderEmail = faker.internet.email();
-        return this.stakeholderEmail;
+        return faker.internet.email();
     }
 
-    create(): void {
+    create({
+        name = this.stakeholderName,
+        email = this.stakeholderEmail,
+        jobfunction = null,
+        groups = [],
+        cancel = false,
+    } = {}): void {
         Stakeholders.clickStakeholders();
-        clickByText("button", "Create new");
-        this.getStakeholderName();
-        this.getStakeholderEmail();
-        this.fillEmail(this.stakeholderEmail);
-        this.fillName(this.stakeholderName);
-        submitForm();
+        clickByText(button, createNewButton);
+        if (cancel) {
+            cancelForm();
+        } else {
+            this.fillEmail(email);
+            this.fillName(name);
+            if (jobfunction) {
+                this.selectJobfunction(jobfunction);
+            }
+            if (groups.length != 0) {
+                this.selectGroups(groups);
+            }
+            submitForm();
+            checkSuccessAlert(
+                commonView.successAlertMessage,
+                `Success! ${name} was added as a stakeholder.`
+            );
+        }
     }
 
-    edit(jobFunction?: boolean, groups?: boolean): void {
+    edit({
+        name = this.stakeholderName,
+        email = this.stakeholderEmail,
+        jobfunction = null,
+        groups = [],
+        cancel = false,
+    } = {}): void {
         Stakeholders.clickStakeholders();
         selectItemsPerPage(100);
         cy.wait(2000);
         cy.get(tdTag)
             .contains(this.stakeholderEmail)
             .parent(trTag)
             .within(() => {
-                click(editButton);
+                click(commonView.editButton);
             });
-        this.getStakeholderName();
-        this.fillName(this.stakeholderName);
-        // jobfunction and group edit to be implemented
-        submitForm();
+        if (cancel) {
+            cancelForm();
+        } else {
+            if (
+                name != this.stakeholderName ||
+                email != this.stakeholderEmail ||
+                jobfunction ||
+                groups.length != 0
+            ) {
+                this.fillName(name);
+                if (email != this.stakeholderEmail) this.fillEmail(email);
+                if (jobfunction) this.selectJobfunction(jobfunction);
+                if (groups.length != 0) this.selectGroups(groups);
+                submitForm();
+                this.stakeholderName = name;
+                this.stakeholderEmail = email;
+            } else {
+                cy.get(commonView.submitButton).should("not.be.enabled");
+            }
```

Comment:
@nitishSr @sshveta We need to discuss over having such assertions in methods.
IMO, Assertions should not be method which are used widely in other test cases. If it fails then it will fail all the other test cases. it will always not find out that as bug in product, sometimes that might be xpath/locator change.

Reply:
I have updated this, I agree that assertion should be kept out of this method.

Reply:
Yes assertions should be in tests